File: SOURCE_EXTERNAL/VecompSoftware.DocSuite.ExternalViewer/src/app/mappers/dossiers/dossier.mapper.ts
import { Injectable } from '@angular/core';
import { BaseMapper } from '../base.mapper';
import { DossierFolderMapper } from './dossier-folder.mapper';
import { DossierModel } from '../../models/dossiers/dossier.model';
import { DossierRoleMapper } from './dossier-role.mapper';

@Injectable()
export class DossierMapper implements BaseMapper {

    constructor(private dossierFolderMapper: DossierFolderMapper, private dossierRoleMapper: DossierRoleMapper) { }

    mapFromJson(json: any): DossierModel {

        if (!json) {
            return null;
        }

        let model: DossierModel = new DossierModel();

        model.uniqueId = json.UniqueId;
        model.year = json.Year;
        model.number = json.Number;
        model.subject = json.Subject;
        model.note = json.Note;
        model.title = json.Title;
        model.registrationUser = json.RegistrationUser;
        model.registrationDate = json.RegistrationDate;
        model.lastChangedUser = json.LastChangedUser;
        model.lastChangedDate = json.LastChangedDate;
        model.startDate = json.StartDate;
        model.endDate = json.EndDate;
        model.containerId = json.ContainerId;
        model.containerName = json.ContainerName;
        model.jsonMetadata = json.JsonMetadata;

        model.dossierFolders = json.DossierFolders ? json.DossierFolders.map(dossierFolder => this.dossierFolderMapper.mapFromJson(dossierFolder)) : null;
        model.dossierRoles = json.Roles ? json.Roles.map(role => this.dossierRoleMapper.mapFromJson(role)) : null;

        return model;
    }

}

File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Mapper.Entity/Dossiers/Implementation/DossierMapper.cs
using VecompSoftware.DocSuiteWeb.Entity.Dossiers;

namespace VecompSoftware.DocSuiteWeb.Mapper.Entity.Dossiers
{
    public class DossierMapper : BaseEntityMapper<Dossier, Dossier>, IDossierMapper
    {
        public override Dossier Map(Dossier entity, Dossier entityTransformed)
        {
            #region [ Base ]
            entityTransformed.Year = entity.Year;
            entityTransformed.Subject = entity.Subject;
            entityTransformed.Note = entity.Note;
            entityTransformed.Number = entity.Number;
            entityTransformed.StartDate = entity.StartDate;
            entityTransformed.EndDate = entity.EndDate;
            entityTransformed.MetadataDesigner = entity.MetadataDesigner;
            entityTransformed.MetadataValues = entity.MetadataValues;
            entityTransformed.DossierType = entity.DossierType;
            entityTransformed.Status = entity.Status;
            #endregion

            return entityTransformed;
        }
    }
}
